### PR TITLE
[JBPM-9636] When a non int value is passed as int to jaxb marshaller

### PR DESCRIPTION
**Pull request by fjtirado**
Throwing marshalling exception if org.kie.server.strict.jaxb.format is
true

**JIRA**: 
[link](https://issues.redhat.com/browse/JBPM-9636)

### CODE REVIEW

**Review by afalhambra — COMMENTED** [review]
Looks good to me, maybe a suggestion in the unit test to make sure current behavior is still in place (regression).

File: kie-server-parent/kie-server-api/src/test/java/org/kie/server/api/marshalling/JAXBMarshallerTest.java [hunk 1]
```diff
@@ -94,4 +96,19 @@ public void testMarshallQueryParam() {
         QueryParam param2 = marshaller.unmarshall(converted, QueryParam.class);
         assertTrue(param2.getValue().get(0) instanceof QueryParam);
     }
+
+    @Test
+    public void testMarshallError() {
+        Marshaller marshaller = MarshallerFactory.getMarshaller(new HashSet<>(), MarshallingFormat.JAXB, getClass()
+                .getClassLoader());
+        assertEquals(13, marshaller.unmarshall("<int-type><value>13</value></int-type>", int.class).intValue());
```
[the remark this answers is not in the document]
Comment (fjtirado) [reply kie-server-parent/kie-server-api/src/test/java/org/kie/server/api/marshalling/JAXBMarshallerTest.java:104]: good point

**Review by afalhambra — APPROVED** [review]
Great job

**Comment by elguardian** [comment]
Jenkins retest this please

**Comment by fjtirado** [comment]
jenkins retest this

**Pull request merged**